ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/MiniProtocol/ChainSync/Client.hs

```diff
@@ -2009,9 +2034,12 @@ invalidBlockRejector tracer version getIsInvalidBlock getCandidate =
         -- tentative
         mapM_ (uncurry disconnect)
           $ firstJust
-                (\hdr -> (hdr,) <$> isInvalidBlock (headerHash hdr))
+                (\hdrWithTime ->
+                   let hdr = hwtHeader hdrWithTime in
+                     (hdr,) <$> isInvalidBlock (headerHash hdr)
+                )
           $ (   case isPipeliningEnabled version of
-                    ReceivingTentativeBlocks    -> drop 1
+                    ReceivingTentativeBlocks    -> drop 1 -- REVIEW: we could also use this opportunity to explain this.
```

Comment:
Does the comment above clear it up?
https://github.com/IntersectMBO/ouroboros-consensus/blob/43e04df72c7fee3ffaf5a191c3c2680377e7e1be/ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/MiniProtocol/ChainSync/Client.hs#L2008-L2012
Maybe we could move the comment directly above the `drop 1`?

Reply:
I added:

```haskell
                    ReceivingTentativeBlocks    ->
                      -- As mentioned in the comment above, if the
                      -- header is tentative we skip the fragment tip,
                      -- dropping the first element.
                      drop 1
```